addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/AlertsCacheableScannerUnitTest.java

```diff
@@ -0,0 +1,331 @@
+
+    // When
+    rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+    // Then
+    assertAlertEvidence("no-store");
+  }
+
+  @Test
+  public void shouldRaiseAlertWithPragmaNoStoreDirectiveInResponseHeader()
+      throws HttpMalformedHeaderException {
+    // Given
+    HttpMessage msg = new HttpMessage();
+    msg.setRequestHeader("GET / HTTP/1.1");
+    msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Pragma: no-store");
+
+    // When
+    rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+    // Then
+    assertAlertEvidence("no-store");
+  }
+
+  @Test
+  public void shouldRaiseAlertWhenCacheControlIsPrivate() throws HttpMalformedHeaderException {
+
+    // Given
+    HttpMessage msg = new HttpMessage();
+    msg.setRequestHeader("GET / HTTP/1.1");
+    msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Cache-Control: must-revalidate, private");
+
+    // When
+    rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+    // Then
+    assertAlertEvidence("private");
+  }
+
+  @Test
+  public void shouldRaiseAlertWhenAuthorizationHeaderWithWrongCacheControlDirectiveUsed()
+      throws HttpMalformedHeaderException {
+    // Given
+    HttpMessage msg = new HttpMessage();
+    msg.setRequestHeader("GET / HTTP/1.1\r\n" + "Authorization: basic\r\n");
+    msg.setResponseHeader(
+        "HTTP/1.1 200 OK\r\n"
+            + "Cache-Control: no-cache\r\n"
+            + "Content-Type: text/xml;charset=UTF-8");
+
+    // When
+    rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+    // Then
+    assertAlertEvidence(HttpHeader.AUTHORIZATION + ":");
+  }
+
+  @Test
+  public void shouldRaiseAlertWhenAuthorizationHeaderWithoutCacheControlUsed()
+      throws HttpMalformedHeaderException {
+    // Given
+    HttpMessage msg = new HttpMessage();
+    msg.setRequestHeader("GET / HTTP/1.1\r\n" + "Authorization: basic\r\n");
+    msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Content-Type: text/xml;charset=UTF-8");
+
+    // When
+    rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+    // Then
+    assertAlertEvidence(HttpHeader.AUTHORIZATION + ":");
+  }
+
+  @Test
+  public void
+      shouldRaiseAlertWhenExpiresMaxAgeAndPublicDirectiveMissingAndStatusNonCacheableByDefault()
+          throws HttpMalformedHeaderException {
+    // Given
+    HttpMessage msg = new HttpMessage();
+    msg.setRequestHeader("GET / HTTP/1.1");
+    msg.setResponseHeader("HTTP/1.1 208 OK\r\n" + "Cache-Control: no-cache");
+    // When
+    rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+    // Then
+    assertAlertEvidence("208");
+  }
+
+  @Test
+  public void shouldRaiseAlertWhenNoCacheDirectiveGiven() throws HttpMalformedHeaderException {
+    // Given
+    HttpMessage msg = new HttpMessage();
+    msg.setRequestHeader("GET / HTTP/1.1");
+    msg.setResponseHeader(
+        "HTTP/1.1 200 OK\r\n"
+            + "Cache-Control: public, no-cache #Content-Type"
```

Comment:
` #Content-Type` ??

`...WhenInvalidCacheDirectiveGiven()`? Then make it some kind of string that really is out of place (ex: `foobar`), as it is I think it might be confusing to someone that has to maintain it in months/years. (Or maybe I'm already misunderstanding it :man_shrugging: )

Reply:
I renamed this method to `shouldRaiseAlertWhenResponseIsStoreableButNotCacheable`, which describes the alert best i think, and removed `#Content-Type`, which is an optional field-name.
Here we must use `no-cache`, because the class under test raises the alert (line 410-433), if the directive `no-cache` is given.